### PR TITLE
Broker server tests

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will fix broken NO_ZMQ tests and fix some issues with broker_server

### CODE REVIEW

File: src/helics/apps/helics-broker-server.cpp [hunk 1]
```diff
@@ -202,11 +218,13 @@ void terminalFunction(std::vector<std::string> args)
...
         ->callback(closeBrokerServer);
-    termProg.add_subcommand("broker", "create a new broker with the given arguments")
+    auto brokersub=termProg.add_subcommand("broker", "create a new broker with the given arguments")
         ->callback(newBroker);
 
+    brokersub->add_option("args", bargs, "arguments for the query");
+
     termProg
-        .add_subcommand("terminate!",
+        .add_subcommand("force_terminate",
```
Comment (phlptp) [inline src/helics/apps/helics-broker-server.cpp:227]: the `!terminate` is not a valid CLI command name

File: src/helics/core/BrokerFactory.hpp [hunk 2]
```diff
@@ -78,6 +78,14 @@ namespace BrokerFactory {
 @param brokerName the name of the broker
 @return a shared_ptr to the Broker*/
     std::shared_ptr<Broker> findBroker(const std::string& brokerName);
+    /** get the first available broker
+@param brokerName the name of the broker
```
Comment (nightlark) [inline src/helics/core/BrokerFactory.hpp:82]:
```suggestion
```

File: src/helics/core/BrokerFactory.cpp [hunk 3]
```diff
@@ -184,7 +184,34 @@ thread which allows the destructor to fire if need be without issue*/
...
+            return brk;
+        }
+        if (brokerName.empty()) {
+            return findBroker();
+        }
+        if (brokerName.front() == '#') {
+            return findBroker(brokerName.substr(1));
+        }
+        try {
+            auto val = std::stoull(brokerName);
```
Comment (nightlark) [inline src/helics/core/BrokerFactory.cpp:198]: Users should probably avoid using findBroker if their goal is to get a broker by index. A broker given a numeric name would prevent finding a broker with the matching numeric index, and a search for a broker named "123abc" or "123xyz" would both return a broker at index 123.

Comment (phlptp) [reply src/helics/core/BrokerFactory.cpp:198]: I am going to change the names of the functions,  but there needed to be a way to get a broker by index from the string to handle some situations in the webserver.

**Review by nightlark — APPROVED** [review]

**Pull request merged**